.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,23 @@
+name: build and test gradle project
+
+on:
+  push:
+    branches-ignore:
+    - master
+    - version-*
+    - dependabot*
+    paths-ignore:
+      - README.md
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project sources
+        uses: actions/checkout@v3
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
```

Comment:
Why do you need to force this? I think the default value was good enough.

Reply:
default value of `cache-read-only` is false only for master branch

Reply:
If we write into the cache on each test build we possibly end up with lots of unused libraries in that case (because we might update versions in feature branches but decide to go with another one in the final version)